controllers/nodeupdate_controller.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/IBM-Cloud/power-go-client/power/models"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/ibm-powervs-block-csi-driver/pkg/cloud"
+)
+
+// NodeUpdateReconciler reconciles a NodeUpdate object
+type NodeUpdateReconciler struct {
+	Client client.Client
+	Scheme *runtime.Scheme
+}
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.10.0/pkg/reconcile
+func (r *NodeUpdateReconciler) Reconcile(_ context.Context, req ctrl.Request) (ctrl.Result, error) {
+
+	// Fetch the Node instance
+	node := corev1.Node{}
+	err := r.Client.Get(context.Background(), req.NamespacedName, &node)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// req object not found, could have been deleted after reconcile req.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			klog.Infof("%s: Node not found - do nothing", req.NamespacedName)
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the req.
+		return ctrl.Result{}, fmt.Errorf("error getting node: %v", err)
+	}
+
```

Comment:
I think if we add a comment with the ProviderID format we are expecting , It will be easy for reader.
Also I think we can add a simple check to confirm that the ProviderID has required fields present, say len(data) == expected length

Reply:
Done.
- comment added
- check for provideId length added